docs/src/reference/message-correlation.md

```diff
@@ -16,6 +16,8 @@ A message is published by the payment service using one of the Zeebe clients. It
 
 > Note that a message can be correlated to multiple workflow instances if they share the same correlation information. But it can be correlated only once per workflow instance.
 
+> In the case of a message subscription opened due to a non-interrupting message boundary event, incoming messages will be correlated as they are published. In rare cases, since execution is interleaved, this may result in messages appearing to be processed out of order.
```

Comment:
I would say the message subscription is more implementation detail, which is not explained in the docs. Furthermore I would remove this section since we not guarantee any ordering in the docs right?

Reply:
With #1769 though this isn't true any more, so I'll just remove it.